### PR TITLE
Add optional validate_records option

### PR DESCRIPTION
**Pull request by koszti**
Adds `validate_records` option:

(Default: False) Validate every single record message to the corresponding JSON schema. This option is disabled by default and invalid RECORD messages will fail only at load time by Snowflake. Enabling this option will detect invalid records earlier but could cause performance degradation.

### CODE REVIEW

File: tests/integration/test_target_snowflake.py [hunk 1]
```diff
@@ -820,3 +821,17 @@ def test_flush_streams_with_intermediate_flushes_on_all_streams(self, mock_emit_
...
+        tap_lines = test_utils.get_test_tap_lines('messages-with-invalid-records.json')
+
+        # Loading invalid records when record validation enabled should fail at ...
+        self.config['validate_records'] = True
+        with assert_raises(RecordValidationException):
+            self.persist_lines_with_cache(tap_lines)
+
+        # Loading invalid records when record validation disabled should fail at load time
+        self.config['validate_records'] = False
+        with assert_raises(Exception):
```
Comment (Samira-El) [inline tests/integration/test_target_snowflake.py]: At load time, shouldn't the raised exception be known? `Exception` is too broad, `with assert_raises(Exception):` will also catch `RecordValidationException`.

Comment (koszti) [reply tests/integration/test_target_snowflake.py]: yes thank you, I'll change it

Comment (koszti) [reply tests/integration/test_target_snowflake.py]: fixed by bc7a35d4bfbe10405bea36e01a754dc0aa879374

Comment (koszti) [reply tests/integration/test_target_snowflake.py]: hmm test failing, I’ll fix it and come back to you with greens :) thanks

**Pull request merged**